### ISSUE TITLE
Multi-day event date time is formatted differently from single-day events with date-fns localizer

### ISSUE THREAD

**Issue by hnerdrum** [jquense/react-big-calendar#2370]
### Check that this is really a bug

- [X] I confirm

### Reproduction link

https://codesandbox.io/s/react-big-calendar-example-forked-w2sbo0?file=/src/App.js

### Bug description

The datetime on multi-day events seems to be formatted differently from single-day events when using the `dateFnsLocalizer`.

As an example, in the locale that is relevant for my use case (`nb-NO`), the datetime in multi-day events is formatted along the lines of `7:30pm.-` on the starting day and something like `- 2:00pm.` on the end day, even though I would expect it to be `07:30. -` and `- 14:00.` respectively.

On single-day events the formatted seems correct according to my expectations

See attached screenshots for examples from the linked code sandbox

| Multi-day (unexpected) | Single-day (as expected |
| ------ | ----- |
| <img width="688" alt="Screenshot 2023-02-20 at 11 08 43" src="https://user-images.githubusercontent.com/25695037/220078190-aaf3dfc5-1f9b-4963-a2f8-21dcdf667560.png">    | <img width="694" alt="Screenshot 2023-02-20 at 11 08 25" src="https://user-images.githubusercontent.com/25695037/220078209-075b3f24-4bb3-4819-8f20-452c51bff1da.png">  |

### Expected Behavior

Multi-day event datetimes are formatted differently from single-day ones.

### Actual Behavior

Multi-day event datetimes should be formatted exactly as single-day ones.

### react-big-calendar version

1.5.0

### React version

18.0.0

### Platform/Target and Browser Versions

macOS Chrome 109.0.5414.119

### Validations

- [X] Read the [docs](http://jquense.github.io/react-big-calendar/examples).
- [X] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.
- [X] Make sure this is a react-big-calendar issue and not an implementation issue

### Would you like to open a PR for this bug?

- [X] I'm willing to open a PR

**Comment by bu6n** [comment]
That's a different format you have to specify:
https://jquense.github.io/react-big-calendar/examples/index.html?path=/docs/props--formats-event-time-range-start-format
https://jquense.github.io/react-big-calendar/examples/index.html?path=/docs/props--formats-event-time-range-end-format

**Comment by hnerdrum** [comment]
> That's a different format you have to specify: https://jquense.github.io/react-big-calendar/examples/index.html?path=/docs/props--formats-event-time-range-start-format https://jquense.github.io/react-big-calendar/examples/index.html?path=/docs/props--formats-event-time-range-end-format

Right, I guess I would just expect the default formatting to be equal for single- and multi-day events without needing to override those properties. Or is there a particular reason they should not be?

**Comment by bu6n** [comment]
No idea.
The default formats are specified here:
https://github.com/jquense/react-big-calendar/blob/master/src/localizers/date-fns.js

I don't know why it's sometimes `p` and sometimes `h:mma`, probably because the original contributor used it this way? A PR is always welcomed!
